Doc/whatsnew/3.14.rst

```diff
@@ -102,25 +102,6 @@ and improvements in user-friendliness and correctness.
 * :ref:`Binary releases for the experimental just-in-time compiler <whatsnew314-jit-compiler>`
 
 
-Incompatible changes
-====================
-
-On platforms other than macOS and Windows, the default :ref:`start
-method <multiprocessing-start-methods>` for :mod:`multiprocessing`
-and :class:`~concurrent.futures.ProcessPoolExecutor` switches from
-*fork* to *forkserver*.
-
-See :ref:`(1) <whatsnew314-concurrent-futures-start-method>` and
-:ref:`(2) <whatsnew314-multiprocessing-start-method>` for details.
-
-If you encounter :exc:`NameError`\s or pickling errors coming out of
-:mod:`multiprocessing` or :mod:`concurrent.futures`, see the
-:ref:`forkserver restrictions <multiprocessing-programming-forkserver>`.
-
-The interpreter avoids some reference count modifications internally when
-it's safe to do so. This can lead to different values returned from
-:func:`sys.getrefcount` and :c:func:`Py_REFCNT` compared to previous versions
-of Python.  See :ref:`below <whatsnew314-refcount>` for details.
```

Comment:
> Fold 'Incompatible changes' into OLC & elsewhere (every change can be incompatible if we try hard enough!)

Added in https://github.com/python/cpython/pull/128173 where we felt it was a big change that needed highlighting. Okay with me to remove if no objections from @picnixz and @gpshead.

Reply:
Maybe we should also mention it in `removed`? My rationale is that whilst this is a notable change for those using multiprocessing on Linux/BSD/etc, it isn't relevant to many other users. As I alluded to, many other changes are incompatible (notably the entire 'Removed' section, but also eg the changes to `finally` which people have complained about, perhaps the new syntax errors with `-O`, etc etc). 

Determining which are special enough to go into a dedicated 'Incompatible Changes' section seems the wrong choice to me, instead we should have confidence in our deprecation process, and note the changes to each module/area in their own relevant place.

A

Reply:
Well.. it *is* an incompatible change and people need to be aware of the reasons. We can be breaking lots of code without them knowing *why* it was actually broken. We didn't have any DeprecationWarning actually. It wasn't possible to do it (see https://github.com/python/cpython/pull/101551 which reverted it). So we could only document it but it wasn't enough.

Reply:
Also, not that the default for macOS is actually 'spawn' and not 'forkserver' and this actually changes a lot. The switch from 'fork' to 'spawn' in macOS was done in 3.8 and it was a smooth transition, but the switch from 'fork' to 'forkserver' on Linux has really been painful. One reason is that people running code on servers usually have Windows or Linux servers, but not macOS servers. So the pickability is actually an issue here.

Reply:
I didn't realise there were no runtime deprecation warnings, that's less ideal.

- The remainder of my argument still holds, though, that this is only one of many incompatible changes in 3.14. Because of our release policy, every version contains incompatible changes -- I see a real risk that by highlighting only some in a special 'Incompatible Changes' section, it diminishes all the others in the user's mind.

The PR/issue you linked noted as its rationale that the runtime warning was showing up for the wrong people and was too disruptive, I would make the same argument here, just relating to prose!

A

Reply:
On the same note, can we actually advertise (if it's not already the case), the "Porting to XXX" section? because it's at the very *bottom* of the page and sometimes people don't look there...